### PR TITLE
Add descriptive MLflow 2.0 migration error message for invalid serving input formats

### PR DESCRIPTION
**Pull request by dbczumar**
Signed-off-by: dbczumar <corey.zumar@databricks.com>

## Related Issues/PRs

 #xxx

## What changes are proposed in this pull request?

Add descriptive MLflow 2.0 migration error message for invalid serving input formats

## How is this patch tested?

- [X] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [X] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [X] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/pyfunc/scoring_server/__init__.py [hunk 1]
```diff
@@ -69,6 +69,15 @@
...
     f"The input must be a JSON dictionary with exactly one of the input fields {SUPPORTED_FORMATS}"
 )
+SCORING_PROTOCOL_CHANGE_INFO = (
+    f"IMPORTANT: If you're seeing this error, you may be querying an MLflow Model Server running"
+    f" version 2.0 or later from an MLflow client running an earlier version (for example, 1.30.0)."
+    f" To resolve the error, either upgrade your MLflow client to a version >= 2.0 or adjust your"
+    f" MLflow Model's requirements file to set the version of MLflow to the same version as the"
+    f" client (for example, change the 'mlflow' requirement string to 'mlflow==1.30.0'). For more"
+    f" information about the updated model scoring server protocol in MLflow 2.0, see "
+    f" https://mlflow.org/docs/latest/models.html#deploy-mlflow-models."
```
Comment (harupy) [inline mlflow/pyfunc/scoring_server/__init__.py:79]:
```suggestion
    "IMPORTANT: If you're seeing this error, you may be querying an MLflow Model Server running"
    " version 2.0 or later from an MLflow client running an earlier version (for example, 1.30.0)."
    " To resolve the error, either upgrade your MLflow client to a version >= 2.0 or adjust your"
    " MLflow Model's requirements file to set the version of MLflow to the same version as the"
    " client (for example, change the 'mlflow' requirement string to 'mlflow==1.30.0'). For more"
    " information about the updated model scoring server protocol in MLflow 2.0, see "
    " https://mlflow.org/docs/latest/models.html#deploy-mlflow-models."
```

**Comment by dbczumar** [comment]
@harupy Thanks for catching unnecessary f-strings. Fixed!

**Review by harupy — APPROVED** [review]
LGTM!

File: mlflow/pyfunc/scoring_server/__init__.py [hunk 2]
```diff
@@ -69,6 +69,17 @@
...
 )
+SCORING_PROTOCOL_CHANGE_INFO = (
+    "IMPORTANT: The MLflow Model scoring protocol has changed in MLflow version 2.0. If you are"
+    " seeing this error, you are likely using an outdated scoring request format. To resolve the"
+    " error, either update your request format or adjust your MLflow Model's requirements file to"
+    " specify an older version of MLflow (for example, change the 'mlflow' requirement specifier"
+    " to 'mlflow==1.30.0'). If you are making a request using the MLflow client"
+    " (e.g. via `mlflow.pyfunc.spark_udf()`), upgrade your MLflow client to a version >= 2.0 in"
+    " order to use the new request format. For more information about the updated MLflow"
+    " Model Server scoring protocol in MLflow 2.0, see "
```
Comment (harupy) [inline mlflow/pyfunc/scoring_server/__init__.py:80]:
```suggestion
    " Model Server scoring protocol in MLflow 2.0, see"
```

nit

**Review by WeichenXu123 — APPROVED** [review]

**Pull request merged**